src/main/kotlin/RanksManager.kt

```diff
@@ -45,8 +46,14 @@ class RanksManager(private val plugin: StrikeTab) : Listener {
         rankList.entries.forEach { (perm, prefix) ->
             val p = event.player
             if (p.hasPermission(perm)) {
-                p.playerListName = prefix + p.name
-                debug { "Set ${p.name}'s tab name to $prefix${p.name} because they had $perm permission" }
+                val format = plugin.config.getString("default-player-format")?: "%prefix%%name%"
+                val translated =
+                    PlaceholderAPI.setPlaceholders(p, format
+                        .replace("%name%", p.name)
+                        .replace("%prefix%", prefix?: "")
+                    )
+                p.playerListName = translated
+                debug { "Set ${p.name}'s tab name to $translated because they had $perm permission" }
                 return
```

Comment:
Should it also set the playerListName even if there's no matching permission (though they could always add a permission everyone has) :thinking:?

Reply:
That's your code and it seems like its working.. wouldn't mess w/ it tbh lol

Reply:
This entire thing was my code before. Previously setting "" prefix would not have changed anything, now that it's possible that there are other placeholders it could change the "non-ranked" behaviour. Anyway, not needed as there's the workaround of adding default rank with a permission everyone has.